Sources/TuistKit/Cache/XCFrameworkBuilder.swift

```diff
@@ -0,0 +1,175 @@
+                                                       derivedDataPath: derivedDataPath.path)
+            simulatorArguments.append(contentsOf: ["-archivePath", simulatorArchivePath!.pathString])
+            simulatorArguments.append(contentsOf: arguments)
+            Printer.shared.print(subsection: "Building \(target.productName) for simulator")
+            try runCommand(simulatorArguments)
+        }
+
+        // Build the xcframework
+        Printer.shared.print(subsection: "Exporting xcframework for \(target.productName)")
+        let xcframeworkPath = outputDirectory.path.appending(component: "\(target.productName).xcframework")
+        let xcframeworkArguments = xcodebuildXcframeworkCommand(deviceArchivePath: deviceArchivePath,
+                                                                simulatorArchivePath: simulatorArchivePath,
+                                                                productName: target.productName,
+                                                                xcframeworkPath: xcframeworkPath)
+        try runCommand(xcframeworkArguments)
+
+        return xcframeworkPath
+    }
+
+    /// Runs the given command.
+    /// - Parameter arguments: Command arguments.
+    fileprivate func runCommand(_ arguments: [String]) throws {
+        if printOutput {
+            try System.shared.runAndPrint(arguments)
+        } else {
+            try System.shared.run(arguments)
+        }
+    }
+
+    /// Returns the arguments that should be passed to xcodebuild to compile for a device on the given platform.
+    /// - Parameter platform: Platform we are compiling for.
+    fileprivate func deviceDestination(platform: Platform) -> String {
+        switch platform {
+        case .macOS: return "osx"
+        default: return "generic/platform=\(platform.caseValue)"
+        }
+    }
+
+    /// Returns the xcodebuild command to generate the .xcframework from the device
+    /// and the simulator frameworks.
+    ///
+    /// - Parameters:
+    ///   - deviceArchivePath: Path to the archive that contains the framework for the device.
+    ///   - simulatorArchivePath: Path to the archive that contains the framework for the simulator.
+    ///   - productName: Name of the product.
+    ///   - xcframeworkPath: Path where the .xcframework should be exported to (e.g. /path/to/MyFeature.xcframework).
+    fileprivate func xcodebuildXcframeworkCommand(deviceArchivePath: AbsolutePath,
+                                                  simulatorArchivePath: AbsolutePath?,
+                                                  productName: String,
+                                                  xcframeworkPath: AbsolutePath) -> [String] {
+        var command = ["xcrun", "xcodebuild", "-create-xcframework"]
+        command.append(contentsOf: ["-framework", deviceArchivePath.appending(RelativePath("Products/Library/Frameworks/\(productName).framework")).pathString])
```

Comment:
`Products/Library/Frameworks` fixed paths are code smell to me, I guess that would break if the Xcode project configuration may define something custom here. Probably not a problem in 99% of cases but maybe you could offer a default + optional override input flag here?

Reply:
> Products/Library/Frameworks fixed paths are code smell to me, I guess that would break if the Xcode project configuration may define something custom here. 

Yeah, some other things might break too if Apple changes something in the API of their tools or the structure of the frameworks. To catch that we have acceptance & integration tests that should fail if that happens.

> Probably not a problem in 99% of cases but maybe you could offer a default + optional override input flag here?

That would not change anything, would it? Instead of having the path defined in this line, it'd be defined a few lines above.

Reply:
I wrote that cause I thought you can provide the product path in the build settings. But I may be wrong and anyway as you say, this can be solved later 👍